### PR TITLE
Implement a machine charm flavor

### LINKED ISSUE #49: Build a machine charm flavor of Grafana Agent

### LINKED ISSUE #130: (machine) Use the `positions_directory` config option
> I played around a little with this. You can also set a folder in which the grafana-agent will generate its own position files based on the job name so you don't have to set the name manually for every job. There's a field on the same level as "configs" called "positions_directory" (see https://grafana.com/docs/agent/latest/configuration/logs-config/). I had to put it in /var/snap/[...] to get it to work permissions wise, but that did the trick for me at least. Have you considered it?

_Originally posted by @awnns in https://github.com/canonical/grafana-agent-k8s-operator/issues/116#issuecomment-1460315955_

### PR DESCRIPTION
**Pull request by dstathis**
closes #49 
closes #130

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -43,10 +43,13 @@ commands =
...
-allowlist_externals = cp
+allowlist_externals =
+    cp
+    chmod
 commands =
-    cp src/machine_charm.py src/charm.py
-    cp machine_metadata.yaml metadata.yaml
+    cp {toxinidir}/src/machine_charm.py {toxinidir}/src/charm.py
+    chmod +x {toxinidir}/src/charm.py
+    cp {toxinidir}/machine_metadata.yaml {toxinidir}/metadata.yaml
```
Comment (PietroPasotti) [inline tox.ini:52]: like charm.py is gitignored, should we also gitignore metadata.yaml?

Comment (PietroPasotti) [reply tox.ini:52]: (and have a separate k8s-metadata.yaml?)

Comment (dstathis) [reply tox.ini:52]: That is exactly what we do. Unfortunately, part of the CI (I don't remember which part) requires a stub metadata.yaml file to exist. That stub being checked in "overrides" gitignore :(

**Comment by rbarry82** [comment]
At least updating libraries squashed some of the conflicts

**Pull request merged**